app/assets/javascripts/pawnPromotion.js

```diff
@@ -0,0 +1,50 @@
+$(document).ready(function() {
+    $("#Queen-promotion").click(function () {
```

Comment:
Lots of duplication here. Why not make a `piece-promotion` class, then ` data-type` with the piece type. Then just use `piece.data('type')` for the `new_piece` declaration